packages/angular/cli/src/commands/mcp/tools/serve.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { ChildProcess, spawn } from 'child_process';
+import { z } from 'zod';
+import { McpToolContext, declareTool } from './tool-registry';
+
+let devServerProcess: ChildProcess | null = null;
+const serverLogs: string[] = [];
+
+const serveToolInputSchema = z.object({
+  command: z.enum(['start_devserver', 'stop_devserver']).describe('The subcommand to execute.'),
+  project: z
+    .string()
+    .optional()
+    .describe(
+      'Which project to serve in a monorepo context. If not provided, serves the top-level project.',
+    ),
+  configuration: z
+    .string()
+    .optional()
+    .default('development')
+    .describe('Which build configuration to use. Defaults to "development".'),
+});
+
+export type ServeToolInput = z.infer<typeof serveToolInputSchema>;
+
+const serveToolOutputSchema = z.object({
+  message: z.string().describe('A message indicating the result of the operation.'),
+  logs: z.array(z.string()).optional().describe('The logs from the dev server.'),
+});
+
+export type ServeToolOutput = z.infer<typeof serveToolOutputSchema>;
+
+function serveToolFactory(context: McpToolContext) {
+  return (input: ServeToolInput) => {
+    switch (input.command) {
+      case 'start_devserver':
+        if (devServerProcess) {
+          return {
+            structuredContent: {
+              message: 'Development server is already running.',
+            },
+          };
+        }
+
+        const args = ['serve'];
+        if (input.project) {
+          args.push(input.project);
+        }
+        if (input.configuration) {
+          args.push(`--configuration=${input.configuration}`);
+        }
+
+        serverLogs.length = 0; // Clear previous logs
+        devServerProcess = spawn('ng', args, { stdio: 'pipe' });
+
+        devServerProcess.stdout?.on('data', (data) => {
+          serverLogs.push(data.toString());
+        });
+        devServerProcess.stderr?.on('data', (data) => {
+          serverLogs.push(data.toString());
+        });
+
+        devServerProcess.on('close', () => {
+          devServerProcess = null;
+        });
+
+        return {
+          structuredContent: {
+            message: 'Development server started.',
+          },
+        };
+
+      case 'stop_devserver':
+        if (!devServerProcess) {
+          return {
+            structuredContent: {
+              message: 'Development server is not running.',
+            },
+          };
+        }
+
+        devServerProcess.kill('SIGTERM');
+        devServerProcess = null;
+
+        return {
+          structuredContent: {
+            message: 'Development server stopped.',
+            logs: serverLogs,
+          },
+        };
```

Comment:
Question: Why does this return `structuredContent` but the `build` tool returns `text`? Is there a meaningful difference between these two?